gendiff/string_templates.py

```diff
@@ -0,0 +1,148 @@
+from typing import Any, Dict
+
+from gendiff.diff import get_diff_view
+
+diffs_templates = {
+    "missing": "{indent}- {key}:{symb}{value}\n",
+    "added": "{indent}+ {key}:{symb}{value}\n",
+    "changed": "{indent}- {key}:{symb}{value}\n{indent}+ {key}: {ch_value}\n",
+    "no_change": "{indent}  {key}:{symb}{value}\n",
+    "nested": "{indent}  {key}: {{\n{value}{indent}  }}\n",
+    "changed_nested": "{indent}{spec_char} {key}: {{\n{value)}{indent}  }}\n",
+    "missing_nested": "{indent}- {key}: {{\n{value}{indent}  }}\n",
+    "added_nested": "{indent}+ {key}: {{\n{value}{indent}  }}\n",
+}
+
+
+def convert_value(val: Any) -> Any:
+    if not isinstance(val, bool | None):
+        return val
+    exceptions_to_convert = {False: "false", True: "true", None: "null"}
+    return exceptions_to_convert[val]
+
+
+def flat_diff(
+    key_type: str,
+    indent: int,
+    key: str,
+    spec_char: str = " ",
+    symb: str = " ",
+    value: Any = None,
+    ch_value: Any = None,
+) -> str:
+    if key_type in ("missing", "added", "no_change"):
+        diff_str = diffs_templates[key_type].format(
+            indent=indent * " ", key=key, symb=symb, value=value
+        )
+    elif key_type == "nested":
+        diff_str = diffs_templates["nested"].format(
+            indent=indent * " ", key=key, value=value
+        )
+    elif key_type == "changed_nested":
+        diff_str = diffs_templates["changed_nested"].format(
+            indent=indent * " ", key=key, spec_char=spec_char, value=value
+        )
+    elif key_type == "missing_nested":
+        diff_str = diffs_templates["missing_nested"].format(
+            indent=indent * " ", key=key, value=value
+        )
+    elif key_type == "added_nested":
+        diff_str = diffs_templates["added_nested"].format(
+            indent=indent * " ", key=key, value=value
+        )
+    else:
+        diff_str = diffs_templates["changed"].format(
+            indent=indent * " ", key=key, symb=symb, value=value, ch_value=ch_value
+        )
+
+    return diff_str
+
+
+def get_diff_str_nested(
```

Comment:
you should have three functions:  
- one that is responsible for the creation of diff (takes care of `{}`)
- one that is responsible for the processing of nested diff types
- one that is responsible for the processing of plain diff types